src/data_visualization/src/scatter_plot.rs

```diff
@@ -0,0 +1,168 @@
+#![allow(dead_code)]
+#![allow(unused)]
+
+use itertools_num::linspace;
+use plotly::common::{
+	self, ColorScale, ColorScalePalette, DashType, Fill, Font, Line, LineShape, Marker, Mode, Title,
+};
+use plotly::layout::{self, Annotation, Axis, BarMode, Layout, Legend, TicksDirection};
+use plotly::{plot, Scatter};
+use plotly::{ImageFormat, Plot};
+
+use plotly::box_plot::{BoxMean, BoxPoints};
+use plotly::color::{NamedColor, Rgb, Rgba};
+use plotly::common::{Anchor, ErrorData, ErrorType, Orientation};
+use plotly::histogram::{Bins, Cumulative, HistFunc, HistNorm};
+use plotly::layout::{BoxMode, Margin, TraceOrder};
+use plotly::{Bar, BoxPlot, Histogram};
+use polars::error::{polars_bail, PolarsError};
+use polars::frame::DataFrame;
+use polars::prelude::{col, lit, IntoLazy};
+use polars::series::Series;
+use rand;
+use rand_distr::{Distribution, Normal, Uniform};
+
+use plotly::common::Side;
+use plotly::layout::ItemSizing;
+use plotly::layout::{GridPattern, LayoutGrid, RowOrder};
+
+use std::{error::Error, process};
+
+fn make_trace(
+	name: &str,
+	i: usize,
+	data: &Series,
+	house_name: &str,
+) -> Result<Box<Scatter<f64, f64>>, Box<dyn Error>> {
+	//
+	let mut col: Vec<f64> = data
+		.f64()?
+		.into_iter()
+		.filter(|x| x.is_some())
+		.map(|x| x.unwrap())
+		.collect();
+
+	col.retain(|x| *x != 0.0);
+	let slice: Vec<&f64> = col.iter().take(3).collect();
+	// println!("{name}: {:?}", slice);
+	if (col.is_empty()) {
+		Err("No float values in column")?;
+	}
+
+	let min = 0.0;
+	let max = col.len() as f64;
+	let n = (max - min) as usize;
+	// col.sort_by(|a, b| a.partial_cmp(b).unwrap());
+	let t: Vec<f64> = col.iter().copied().rev().collect();
+	// let t = (&t[0..t.len() / 2]).to_vec();
+	let y = col;
+	// let y = (&y[0..y.len() / 2]).to_vec();
+
+	let color = match house_name {
+		"Gryffindor" => NamedColor::Red,
+		"Ravenclaw" => NamedColor::Blue,
+		"Slytherin" => NamedColor::Green,
+		"Hufflepuff" => NamedColor::Orange,
+		_ => NamedColor::Black,
+	};
+
+	Ok(Scatter::new(t, y)
+		.mode(Mode::Markers)
+		.marker(Marker::new().color(color).size(3))
+		.x_axis(format!("x{}", i))
+		.y_axis(format!("y{}", i)))
+}
+
+fn write_trace(
+	name: &str,
+	house_name: &str,
+	i: usize,
+	one_house: DataFrame,
+	legend_check: &mut usize,
+	plot: &mut Plot,
+	check: &mut bool,
+	layout: &mut Layout,
+) {
+	let series = one_house.column(name).unwrap();
+	let annot = Annotation::new()
+		.y(1.1)
+		.x(0.5)
+		.y_ref(format!("y{} domain", i))
+		.x_ref(format!("x{} domain", i))
+		.y_anchor(Anchor::Top)
+		.x_anchor(Anchor::Center)
+		.show_arrow(false)
+		.text(format!("{}", name));
+
+	match make_trace(name, i, series, house_name) {
+		Err(e) => {
+			println!("column error: {}", e);
+			()
+		}
```

Comment:
return the error instead of printing it

+ `println!` prints on `STDOUT` instead of `STDERR` (`eprintln!` prints on `STDERR`)